lib/rack/request.rb

```diff
@@ -398,8 +398,8 @@ def forwarded_authority
               return forwarded.last
             end
           when :x_forwarded
-            if value = get_header(HTTP_X_FORWARDED_HOST)
-              return wrap_ipv6(split_header(value).last)
+            if x_forwarded_host = split_header(get_header(HTTP_X_FORWARDED_HOST)).last
```

Comment:
```suggestion
            if (value = get_header(HTTP_X_FORWARDED_HOST)) && (x_forwarded_host = split_header(value).last)
```

This avoids an unnecessary array allocation if the header is not set.

Reply:
Thank you! I've merged it.